sox/transform.py

```diff
@@ -1820,6 +1826,55 @@ def mcompand(self, n_bands=2, crossover_frequencies=[1600],
         self.effects_log.append('mcompand')
         return self
 
+    def noiseprof(self, profile='noise.prof'):
+        '''Calculate  a  profile  of  the audio for use in noise reduction.
+        See the description of the noisered effect for details.
+
+        Paramters
+        ---------
+        profile : str, default='noise.prof'
+            A temp file used to store noise inforation
+
+        See Also
+        --------
+        noisered
+        '''
+        effect_args = ['noiseprof', profile]
+
+        self.effects.extend(effect_args)
+        self.effects_log.append('noiseprof')
+        return self
+
+    def noisered(self, amount=0.5, profile='noise.prof'):
+        '''Reduce noise in the audio signal  by  profiling  and  filtering.
+        This effect is moderately effective at removing consistent back‐
+        ground noise such as hiss or hum.
+
+        Paramters
+        ---------
+        amount : float, default=0.5
+            How much noise should be removed is specified by amount-a number
+            between  0  and  1  with  a default of 0.5.  Higher numbers will
+            remove more noise but present a greater likelihood  of  removing
+            wanted  components  of  the  audio  signal.
+        profile: str, default='noise.prof'
+            The noise data generated by noiseprof
+        '''
+        if not isinstance(amount, float):
```

Comment:
Since the number has to be between 0 and 1, let's check that here too. Also, the function shouldn't fail if the user passes an int (for example `1`):

```python
 if not is_number(amount) or amount < 0 or amount > 1:
    raise ValueError("amount must be a number between 0 and 1.")
```